### DISCUSSION TITLE
Proposal: A modern website

### DISCUSSION THREAD

**Discussion by securitykernel** [randombit/botan#3834] [Show and tell]
We propose a new, modern website for Botan that mainly serves as an entry point for new users and is served under a dedicated top-level domain as other cryptographic libraries do, e.g., OpenSSL's [openssl.org](https://openssl.org), BouncyCastle's [bouncycastle.org](https://bouncycastle.org) or GnuTLS' [gnutls.org](https://www.gnutls.org).

We have registered [https://botan-crypto.org](https://botan-crypto.org) to showcase the new website. https://botan.org is already registered and in use and therefore not available.

_For discussion it is best to just browse the website now. For historical references I'll also add screenshots below._

The website should consist of three major areas:

# Landing Page

Then idea here is to provide a short summary on what Botan is, and quick links to GitHub, to the getting started guide and the docs in general. The docs are hosted on the subdomain https://docs.botan-crypto.org, so essentially this reflects what is currently hosted on https://botan.randombit.net, except for releases.

Additionally, the landing page should highlight what sets Botan apart from other cryptographic libraries. I gave this a shot already, any feedback is welcome!

![image](https://github.com/randombit/botan/assets/8866185/ed54f755-9500-4962-9257-cd9f0da5cf81)

# Blog

Regularly post about technical topics involving Botan beyond what is happening on GitHub. @reneme already posted 
["Using Botan’s TLS with Boost Beast"](https://botan-crypto.org/using-botan-tls-with-boost-beast/), additional topics he proposed are:

* Botan's Module System: Pay what you use!
* Benchmark: How does Botan compare to OpenSSL?
* BSI Project P.481
* Hybrid Key Exchange in TLS 1.3

![image](https://github.com/randombit/botan/assets/8866185/34bd8228-312a-484a-9960-a8bfc80374be)

# PQC

PQC is one of *the* major topics in crypto for the next 10-15 years, and in Botan a lot is happening in this area now thanks to BSI Project 481 (#3108), so It's fair to give it a separate page with updates on PQC support & progress in Botan addressing algorithms as well as integration into TLS. We could also highlight for each algorithms for which purposes it is suitable or not (think: large keys, large signatures, etc.) The page also links to our PQC/T test server running at https://pq.botan-crypto.org running a `botan tls_server` for compatibility tests with other implementations.

![image](https://github.com/randombit/botan/assets/8866185/652e3e35-d529-4091-8e09-6f29f1240c62)

# Support (optional)

Another page currently not existing is a Support page highlighting where to get support for open source (GH issues, basically), or for commercial purposes (maybe a separate mailbox @botan-crypto.org).

## Technical/Administration

Technically, the website is made with [Wordpress](https://wordpress.org) and the [Blocksy (free) theme](https://creativethemes.com/blocksy/). It is set up to auto-update Wordpress itself and all plugins, so the maintenance effort is really minimal.

What do you think?

**Comment by pstanisz** [comment]
Sounds like a good idea!
